pkg/controller/workload/job/job_controller.go

```diff
@@ -433,27 +482,49 @@ func ConstructWorkloadFor(ctx context.Context, client client.Client,
 }
 
 func appendFinishedConditionIfNotExists(conds []metav1.Condition, jobStatus batchv1.JobConditionType) ([]metav1.Condition, bool) {
+	return appendConditionIfNotExists(conds, generateFinishedCondition(jobStatus))
+}
+
+func generateFinishedCondition(jobStatus batchv1.JobConditionType) metav1.Condition {
```

Comment:
If this function is not used anywhere else, just make it part of appendFinishedConditionIfNotExists

Reply:
Done